src/UiPath.Workflow/Activities/ScriptingJitCompiler.cs

```diff
@@ -36,6 +39,133 @@ public record ExpressionToCompile(string Code, IReadOnlyCollection<string> Impor
     : CompilerInput(Code, ImportedNamespaces)
 { }
 
+public sealed class CachedMetadataReferenceResolver : MetadataReferenceResolver
+{
+    public static CachedMetadataReferenceResolver Default = new CachedMetadataReferenceResolver(ScriptMetadataResolver.Default);
+
+    ScriptMetadataResolver _resolver;
+
+    private class ResolveCacheKey
```

Comment:
Can you use a record for this instead? It already has equals and hashcode implementations that do very similar things to what you're doing here.